### ISSUE TITLE
[Release] Tags should be validated

### ISSUE THREAD

**Issue by itsjustkevin** [flutter/flutter#128792]
Release tags should be validated before pushing through the MPA process.  

Without release tag validation, we could end up in a situation where our tag name is malformed, causing issues with the flutter tool.

**Comment by itsjustkevin** [comment]
CC: @XilaiZhang @Jasguerrero @CaseyHillers

**Comment by XilaiZhang** [comment]
Yeah sorry I didn't look closely enough when approving the mpa request

**Comment by itsjustkevin** [comment]
The goal is to remove as much possibility of human error as possible.  

Also tagging @sealesj for context.

**Comment by sealesj** [comment]
@itsjustkevin  Are you wanting some sort of automated validation process? There are automated checks in the MPA process for valid formatting of tags, but this won't address the problem of a tag with the correct formatting but incorrect numbering

**Comment by itsjustkevin** [comment]
@sealesj I am assuming there is some way that we can ensure the tag names are validated based on the release channel.

```
release_channel == stable ? stable_tag_regex : beta_tag_regex
```

Or something of the sort to ensure pushing fails if a malformed tag is provided.

**Comment by Jasguerrero** [comment]
@itsjustkevin We should discuss more where is the appropriate fix for this, since we can also add the regex into the recipes

**Comment by sealesj** [comment]
@itsjustkevin Yes, we currently have this check in place, but we could add the additional guardrail of the release channel check. I think it would make sense for that check to be added here https://flutter.googlesource.com/recipes/+/refs/heads/main/recipes/release/release_publish.py#30

**Comment by XilaiZhang** [comment]
From triage: assigning to Godofredo

**Comment by godofredoc** [comment]
@sealesj we can probably solve this with a regex in the recipe. Would you mind taking a look?

**Comment by sealesj** [comment]
Creating CL now

**Comment by sealesj** [comment]
Addressing this problem with: https://flutter-review.googlesource.com/c/recipes/+/46000
